### PR TITLE
add multiregression custom metric

### PR DESCRIPTION
**Pull request by azikmsu**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/

### CODE REVIEW

File: catboost/python-package/catboost/_catboost.pyx [hunk 1]
```diff
@@ -108,6 +108,8 @@ class _NumpyAwareEncoder(JSONEncoder):
             return obj.tolist()
         return JSONEncoder.default(self, obj)
 
+class TMultiRegressionMetric:
```
Comment (nikitxskv) [inline catboost/python-package/catboost/_catboost.pyx]: Let's rename TMultiRegressionMetric MultiLabelCustomMetric here.
And in c++ too (TMultiLabelCustomMetric).

And enumerate all methods which needs to implement.

File: catboost/python-package/catboost/core.py [hunk 2]
```diff
@@ -95,7 +96,20 @@ def get_catboost_bin_module():
...
+        pass
 
+    @abstractmethod
+    def is_max_optimal(self):
+        pass
+    
+    @abstractmethod
+    def get_final_error(self, error, weight):
+        pass
+"""
```
Comment (nikitxskv) [inline catboost/python-package/catboost/core.py]: Remove it

File: catboost/python-package/ut/medium/test.py [hunk 3]
```diff
@@ -264,6 +265,57 @@ def load_simple_dataset_as_lists(is_test):
...
+
+    model = CatBoostRegressor(loss_function='MultiRMSE', iterations=niter, use_best_model=True, eval_metric=MultiRMSE())
+    model.fit(train_pool, eval_set=test_pool)
+    pred1 = model.predict(test_pool)
+
+    model2 = CatBoostRegressor(loss_function='MultiRMSE', iterations=niter, use_best_model=True, eval_metric="MultiRMSE")
+    model2.fit(train_pool, eval_set=test_pool)
+    pred2 = model2.predict(test_pool)
+
+    assert np.all(pred1 == pred2)
```
Comment (nikitxskv) [inline catboost/python-package/ut/medium/test.py]: Need to compare model1.evals_result_ with model2.evals_result_.

File: catboost/python-package/catboost/_catboost.pyx [hunk 4]
```diff
@@ -112,6 +112,18 @@ class _NumpyAwareEncoder(JSONEncoder):
 class CatBoostError(Exception):
     pass
 
+
+class MultiLabelCustomMetric:
+    def evaluate(self):
```
Comment (nikitxskv) [inline catboost/python-package/catboost/_catboost.pyx]: Let's enumerate all params here like in get_final_error method and write docstrings for all these parameters like [here](https://github.com/catboost/catboost/blob/master/catboost/python-package/catboost/core.py#L4507-L4520).
Don't forget to do the following:
```
@cython.embedsignature(True) 
class MultiLabelCustomMetric:
```

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1180295](https://a.yandex-team.ru/review/1180295)

**Comment by arcadia-devtools** [comment]
Check [1180295](https://a.yandex-team.ru/review/1180295) passed
